### PR TITLE
Typing users view model

### PR DESCRIPTION
**Pull request by grzesiekko**
### What?

Typing users view model.

### Why?

.

----

### CODE REVIEW

File: Chatkit/View Models/TypingUsersViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    
+    // MARK: - Private methods
+    
+    private func reload() {
+        guard self.provider.typingUsers.count > 0 else {
+            self.value = nil
+            return
+        }
+        
+        let names = self.provider.typingUsers.map { $0.name ?? "x" }.sorted().joined(separator: ", ")
```
Comment (mdpye) [inline Chatkit/View Models/TypingUsersViewModel.swift]: `"x"` -> `self.userNamePlaceholder`?

File: Chatkit/View Models/TypingUsersViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+/// A view model which provides a text representations of typing users.
+public class TypingUsersViewModel {
+    
+    // MARK: - Properties
+    
+    /// The text representations of typing users.
+    public private(set) var value: String?
+    
+    /// The typing users provider used as the source of data.
+    public let provider: TypingUsersProvider
```
Comment (mdpye) [inline Chatkit/View Models/TypingUsersViewModel.swift]: Surely this is private? What's the benefit in exposing it?

File: Chatkit/View Models/TypingUsersViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    
+    /// The placeholder used when a user doeas not have a value set for the name property.
+    public let userNamePlaceholder: String
```
Comment (mdpye) [inline Chatkit/View Models/TypingUsersViewModel.swift]: Surely this is private? What's the benefit in exposing it?

File: Chatkit/View Models/TypingUsersViewModel.swift (hunk 2 above)
Comment (grzesiekko) [reply Chatkit/View Models/TypingUsersViewModel.swift]: Well it has been injected into the class, it is being retained by the class, so it make sense for this to be exposed as `public`. It is consistent with other SDKs. It also improves testability.

File: Chatkit/View Models/TypingUsersViewModel.swift (hunk 3 above)
Comment (grzesiekko) [reply Chatkit/View Models/TypingUsersViewModel.swift]: Please see the comment above.

File: Chatkit/View Models/TypingUsersViewModel.swift (hunk 1 above)
Comment (grzesiekko) [reply Chatkit/View Models/TypingUsersViewModel.swift]: Fixed.

File: Chatkit/View Models/TypingUsersViewModel.swift (hunk 2 above)
Comment (grzesiekko) [reply Chatkit/View Models/TypingUsersViewModel.swift]: After the discussion the property is now marked as `private`.

File: Chatkit/View Models/TypingUsersViewModel.swift (hunk 3 above)
Comment (grzesiekko) [reply Chatkit/View Models/TypingUsersViewModel.swift]: After the discussion the property is now marked as `private`.

**Pull request merged**